File: modules/core/src/main/java/fr/rozanc/cocozzelle/datatable/DefaultColumnExpression.java
package fr.rozanc.cocozzelle.datatable;

import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.Map;

public class DefaultColumnExpression implements ColumnExpression {

    private final String originalString;

    private final String expressionString;

    private final Map<String, String> options;

    public DefaultColumnExpression(final String originalExpression,
                                   final String expression,
                                   final Map<String, String> options) {
        this.originalString = originalExpression;
        this.expressionString = expression;
        this.options = new LinkedHashMap<>(options);
    }

    @Override
    public String getOriginalString() {
        return originalString;
    }

    @Override
    public String getString() {
        return expressionString;
    }

    @Override
    public Map<String, String> getOptions() {
        return Collections.unmodifiableMap(options);
    }
}

File: modules/core/src/main/kotlin/fr/rozanc/kokozzelle/datatable/DefaultColumnExpression.kt
package fr.rozanc.kokozzelle.datatable

class DefaultColumnExpression(
    override val originalString: String,
    override val string: String,
    override val options: Map<String, String>
) : ColumnExpression {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as DefaultColumnExpression

        if (originalString != other.originalString) return false
        if (string != other.string) return false
        if (options != other.options) return false

        return true
    }

    override fun hashCode(): Int {
        var result = originalString.hashCode()
        result = 31 * result + string.hashCode()
        result = 31 * result + options.hashCode()
        return result
    }

    override fun toString(): String {
        return "DefaultColumnExpression(originalString='$originalString', expressionString='$string', options=$options)"
    }
}
